From 11e743a1d630828905ab1db1acd7067e64303297 Mon Sep 17 00:00:00 2001
From: Tiberiu Breana <andrei-tiberiu.breana@nxp.com>
Date: Fri, 23 Jun 2017 17:15:24 +0300
Subject: [PATCH 2074/5242] MLK-15351: PCI: imx: Only use pcie_bus_regulator
 for iMX6QP

commit  4ca2b2f6fcc13e73ed11ddd0c72680eada2a8525 from
https://source.codeaurora.org/external/imx/linux-imx.git

The pcie_bus_regulator is only used by the iMX6QP board,
so only request the regulator for this variant.

Signed-off-by: Tiberiu Breana <andrei-tiberiu.breana@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |   10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index 28585b3..60940d6 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -1840,10 +1840,14 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 		return PTR_ERR(imx6_pcie->pcie);
 	}
 
-	imx6_pcie->pcie_bus_regulator = devm_regulator_get(dev,
-			"pcie-bus");
-	if (IS_ERR(imx6_pcie->pcie_bus_regulator))
+	if (imx6_pcie->variant == IMX6QP) {
+		imx6_pcie->pcie_bus_regulator = devm_regulator_get(dev,
+				"pcie-bus");
+		if (IS_ERR(imx6_pcie->pcie_bus_regulator))
+			imx6_pcie->pcie_bus_regulator = NULL;
+	} else {
 		imx6_pcie->pcie_bus_regulator = NULL;
+	}
 
 	/* Grab GPR config register range */
 	if (imx6_pcie->variant == IMX7D) {
-- 
1.7.9.5

